test/e2e/common/security_context.go

```diff
@@ -371,7 +371,7 @@ var _ = framework.KubeDescribe("Security Context", func() {
 
 // waitForFailure waits for pod to fail.
 func waitForFailure(f *framework.Framework, name string, timeout time.Duration) {
-	gomega.Expect(e2epod.WaitForPodCondition(f.ClientSet, f.Namespace.Name, name, "success or failure", timeout,
+	gomega.Expect(e2epod.WaitForPodCondition(f.ClientSet, f.Namespace.Name, name, "succeeded or failed", timeout,
```

Comment:
let's use the constants everywhere:
`..., name, fmt.Sprintf("%s or %s", v1.PodSucceeded, v1.PodFailed)`, timeout ...`

Reply:
I didn't clearly get what you are referring to. Could you please explain a bit more?

Reply:
instead of using the literal string `succeeded or failed` use `fmt.Sprintf("%s or %s", v1.PodSucceeded, v1.PodFailed)`

Reply:
But in this case there are no two separate strings. "succeeded or failed" is a single string. Do you want me to implement this else where? where these strings are being used separately?

Reply:
Got it Thanks.

Reply:
WaitForPodCondition this function currently accepts single reason as an argument. somecases we have one reason like v1. PodReasonUnschedulable and some cases we have two reasons like v1.PodSucceeded, v1.PodFailed. I am thinking to use variadic function and get the values. Any feedback? as I am new to go.

Reply:
https://github.com/kubernetes/kubernetes/blob/305377a2f577a47e4decc6ebfc4a3c2a393a00f5/test/e2e/framework/pod/wait.go#L211

it's used as `desc` (or description), so it's not exactly a list of reasons or conditions. just a general string.
let's leave it like that and use this `fmt.Sprintf("%s or %s", v1.PodSucceeded, v1.PodFailed)` for your PR.

Reply:
Other statements calling this function with single pod phase:
eg: 	`return WaitForPodCondition(c, namespace, podName, v1.PodReasonUnschedulable, podStartTimeout, func(pod *v1.Pod) (bool, error) {`

**Three Questions and Solutions:**

- Can we use the const `PodPending PodPhase = "Pending"` instead of "Success or Failure"
- Or Can we create new constant 'PodSpawning' ??
- Or Can I use this code for that piece: `podStatus := string(v1.PodSucceeded + " or " + v1.PodFailed)` and use this variable as an argument

Please suggest.

Reply:
@sureshpalemoni you can just make the change i suggested here https://github.com/kubernetes/kubernetes/pull/87598#discussion_r372448051
for this PR.

other changes feel out of scope for this PR.

Reply:
@neolit123 Thank you very much. submitted another commit.